engine/src/main/scala/cromwell/engine/db/ExecutionInfosByExecution.scala

```diff
@@ -0,0 +1,101 @@
+package cromwell.engine.db
+
+import cromwell.engine._
+import cromwell.engine.backend._
+import cromwell.engine.db.slick.{ExecutionInfo, Execution}
+import wdl4s.values.WdlFile
+
+case class ExecutionInfosByExecution(execution: Execution, executionInfos: Seq[ExecutionInfo]) {
+
+  lazy val callLogs: Option[CallLogs] = {
+    import ExecutionInfosByExecution._
+
+    case class Accumulator(stdout: Option[WdlFile], stderr: Option[WdlFile], logs: Map[String, WdlFile])
+
+    val acc = executionInfos.foldLeft(Accumulator(None, None, Map.empty)) { (acc, executionInfo) =>
+      executionInfo.key.split("_", 2) match {
+        case Array(CallLogPrefix, StdoutSuffix) => acc.copy(stdout = executionInfo.value map {
+          WdlFile(_)
+        })
+        case Array(CallLogPrefix, StderrSuffix) => acc.copy(stderr = executionInfo.value map {
+          WdlFile(_)
+        })
+        case Array(CallLogPrefix, callLogName) =>
+          executionInfo.value match {
+            case Some(value) => acc.copy(logs = acc.logs + (callLogName -> WdlFile(value)))
+            case None => acc
+          }
+        case _ => acc
+      }
+    }
+
+    for {
+      stdout <- acc.stdout
+      stderr <- acc.stderr
+    } yield CallLogs(stdout, stderr, Option(acc.logs).filterNot(_.isEmpty))
```

Comment:
By "warn", we could throw an exception instead. It does qualify possibly as an illegal state.